src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs

```diff
@@ -107,9 +107,25 @@ public static void CreateDirectory(string fullPath)
 
             int count = stackDir.Count;
 
-            // If we were passed a DirectorySecurity, convert it to a security
-            // descriptor and set it in he call to CreateDirectory.
-            Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = default;
+            Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs;
+            if (securityDescriptor == null)
+            {
+                secAttrs = default;
+            }
+            else
+            {
+                unsafe
+                {
+                    fixed (byte* pSecurityDescriptor = securityDescriptor)
```

Comment:
This will fail. As soon as you leave the fixed statement the byte array will unpin. If the garbage collector runs the array will move and you'll get an AV or worse.

All of the logic that accesses SECURITY_ATTRIBUTES must be in the fixed block.

Additionally you do not need to check if the passed in array is null. Just pin it and you'll get a null pointer, which is what you want anyway.

Reply:
Note that it was technically incorrect to not set the size before.

Reply:
Ok I'll fix it.